XenAdmin/Dialogs/VMProtectionRecovery/PolicyHistory.cs

```diff
@@ -83,6 +83,22 @@ public void RefreshTab(IVMPolicy policy)
             else
             {
                 comboBox1.Enabled = true;
+                int hoursFromNow = 0;
+                switch (comboBox1.SelectedIndex)
+                {
+                    case 0: /* default value*/
+                        break;
+                    case 1:
+                        hoursFromNow = 24;
+                        break;
+                    case 2:
+                        hoursFromNow = 7 * 24;
+                        break;
+                }
+        
+                PureAsyncAction action = _policy.getAlertsAction(_policy, hoursFromNow);
+                action.Completed += action_Completed;
+                action.RunAsync();
                 RefreshGrid(_policy.PolicyAlerts);

```

Comment:
RefreshGrid() call (which populates the grid with the last 10 messages- not always the correct filter) is not needed here because it is done action_Completed()